source/falaise/testing/test_falaise_tags.cxx

```diff
@@ -0,0 +1,124 @@
+#include <falaise/resource.h>
+#include <falaise/version.h>
+#include <falaise/tags.h>
+
+int main(int /* argc_ */, char** /* argv_ */) {
+  falaise::initialize();
+  int error_code = EXIT_SUCCESS;
+  try {
+    std::clog << "Test program for the 'Falaise' library." << std::endl;
+
+    std::clog << "Falaise version : " << falaise::version::get_version() << std::endl;
+    std::clog << "Tag categories : " << std::endl;
+    std::clog << " - Experiment:          : '" << falaise::tags::experiment_category() << "'"
+              << std::endl;
+    std::clog << " - Experimental setup   : '" << falaise::tags::experimental_setup_category() << "'"
+              << std::endl;
+    std::clog << " - Geometry setup       : '" << falaise::tags::geometry_setup_category() << "'"
+              << std::endl;
+    std::clog << " - Device setup         : '" << falaise::tags::device_setup_category() << "'"
+              << std::endl;
+    std::clog << " - Electronics setup    : '" << falaise::tags::electronics_setup_category() << "'"
+              << std::endl;
+    std::clog << " - CMS setup            : '" << falaise::tags::cms_setup_category() << "'"
+              << std::endl;
+    std::clog << " - Simulation setup     : '" << falaise::tags::simulation_setup_category() << "'"
+              << std::endl;
+    std::clog << " - Reconstruction setup : '" << falaise::tags::reconstruction_setup_category() << "'"
+              << std::endl;
+    std::clog << " - Variant service      : '" << falaise::tags::variant_service_category() << "'"
+              << std::endl;
+    std::clog << " - Services             : '" << falaise::tags::services_category() << "'"
+              << std::endl;
+    std::clog << " - Configuration        : '" << falaise::tags::generic_configuration_category() << "'"
+              << std::endl;
+    std::clog << " - Data                 : '" << falaise::tags::generic_data_category() << "'"
+              << std::endl;
+    std::clog << " - Image                : '" << falaise::tags::generic_image_category() << "'"
+              << std::endl;
+    std::clog << " - Software             : '" << falaise::tags::software_category() << "'"
+              << std::endl;
+    std::clog << " - Firmware             : '" << falaise::tags::firmware_category() << "'"
+              << std::endl;
+    std::clog << " - Hardware             : '" << falaise::tags::hardware_category() << "'"
+              << std::endl;
+
+    datatools::kernel & dtk = datatools::kernel::instance();
+    const datatools::urn_query_service & dtkUrnQuery = dtk.get_urn_query();
+
+    {
+      datatools::urn_info urnInfo;
+      std::string urn("urn:snemo:demonstrator");
+      if (dtkUrnQuery.check_urn_info(urn, falaise::tags::experiment_category())) {
+        urnInfo = dtkUrnQuery.get_urn_info(urn);
+        std::clog << "The tag '" << urn << "' is registered with category '"
+                  << urnInfo.get_category()
+                  << "'." << std::endl;
+      }
```

Comment:
For this and the remaining blocks, what are the success/fail cases? For example, is it required that the `check_urn_info` always returns `true`? What value is expected to be returned by `get_category`? 

A basic smoke test is fine in this PR, and obviously unexpected errors are handled by the overriding `try/catch` block. In future, these tests should prefer the use of the supplied Catch infrastructure both to ease the writing of test and clarify the expected/design behaviour and conditions.